### ISSUE TITLE
Pretty output fails if before hook is run twice

### ISSUE THREAD

**Issue by kduret** [badeball/cypress-cucumber-preprocessor#1091]
### Current behavior

If the base url is updated and is visited in a before hook, then the hook is run twice

`support/e2e.ts`
```
before(() => {
  Cypress.config('baseUrl', 'http://0.0.0.0:4000');

  cy.intercept('/waiting-page', {
    headers: { 'content-type': 'text/html' },
    statusCode: 200
  }).visit('/waiting-page');
});
```

 result :
```
An error was thrown in your plugins file while executing the handler for the before:spec event.

The error we received was:

Error: Unexpected state in beforeSpecHandler: before-spec (this might be a bug, please report at https://github.com/badeball/cypress-cucumber-preprocessor)
    at createError (/home/kduret/workspace/cypress-cucumber-preprocessor/examples/webpack-ts/node_modules/.pnpm/@badeball+cypress-cucumber-preprocessor@18.0.5_pjwvil4ofno6sgsjgd5tdc3zby/node_modules/@badeball/cypress-cucumber-preprocessor/dist/helpers/error.js:9:12)
    at beforeSpecHandler (/home/kduret/workspace/cypress-cucumber-preprocessor/examples/webpack-ts/node_modules/.pnpm/@badeball+cypress-cucumber-preprocessor@18.0.5_pjwvil4ofno6sgsjgd5tdc3zby/node_modules/@badeball/cypress-cucumber-preprocessor/dist/plugin-event-handlers.js:189:43)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
 ELIFECYCLE  Command failed with exit code 1.
```

this seems related to https://github.com/cypress-io/cypress/issues/26300

### Desired behavior

Waiting a fix on cypress, handle this case, handle this specific case in event handler
This has already been done for a similar issue : https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/lib/plugin-event-handlers.ts#L446-L450

### Test code to reproduce

Issue is reproduced here : https://github.com/kduret/cypress-cucumber-preprocessor/tree/reproduce-twice-before/examples/webpack-ts
I run the test with : `pnpm run cypress:run`

### Versions

* **Cypress version**: 13.1.0
* **Preprocessor version**: 18.0.5
* **Node version**: v18.17.0

### Checklist

- [x] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [x] I've read [instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#bug-reports).
- [x] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by badeball** [comment]
What’s your reason for configuring this in the runtime?

**Comment by kduret** [comment]
Currently, cypress does not allow to bypass the health check of the baseUrl when it starts
But we need to start the app during the tests because we have specific cases where the webserver is not available : 
* fresh install of the app
* upgrade of the app

I am aware of this : https://docs.cypress.io/guides/references/best-practices#Web-Servers
But it does not match our use cases

**Comment by badeball** [comment]
By health check, are you referring to the following warning? This isn't a warning you strictly need to pay attention to.

![2023-09-08-182045_5120x1440_scrot](https://github.com/badeball/cypress-cucumber-preprocessor/assets/730419/6f46276a-5eb1-48fd-91bd-e2288ce3d935)

**Comment by kduret** [comment]
Indeed, I'm referring to this warning, but this is blocking when using `cypress:run` : 
```
> @ cypress:run ~/cypress-cucumber-preprocessor/examples/webpack-ts
> cypress run --config-file cypress.config.ts --browser=chrome

Cypress could not verify that this server is running:

  > https://localhost:1311

We are verifying this server because it has been configured as your baseUrl.

Cypress automatically waits until your server is accessible before running tests.

We will try connecting to it 3 more times...
We will try connecting to it 2 more times...
We will try connecting to it 1 more time...

Cypress failed to verify that your server is running.

Please start this server and then run Cypress again.
 ELIFECYCLE  Command failed with exit code 1.
```

It seems there is no way (currently) to bypass this check : https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/server-base.ts#L283-L296

**Comment by badeball** [comment]
I've fixed this with v18.0.6.

**Comment by kduret** [comment]
Thanks ! ❤️
